File: src/main/java/raccoonlang/parser/ast/ClassBodyAstNode.java
package raccoonlang.parser.ast;

import raccoonlang.parser.Parser;
import raccoonlang.tokenizer.TokenType;

import java.util.ArrayList;
import java.util.List;

public class ClassBodyAstNode {

    public List<ClassMemberAstNode> classMemberAstNodes = new ArrayList<>();

    public static ClassBodyAstNode parse(Parser parser) {
        ClassBodyAstNode classBodyAstNode = new ClassBodyAstNode();

        parser.take(TokenType.OPEN_CURLY);

        while(true){
            ClassMemberAstNode classMemberAstNode = ClassMemberAstNode.tryParse(parser);
            if(classMemberAstNode != null){
                classBodyAstNode.classMemberAstNodes.add(classMemberAstNode);
            }else{
                break;
            }
        }

        parser.take(TokenType.CLOSE_CURLY);

        return classBodyAstNode;
    }
}

File: Parser/AST/ClassBodyAstNode.cs
namespace Raccoonlang.Parser.AST;

using Tokenizer;

public class ClassBodyAstNode
{
    public List<ClassMemberAstNode> memberList = new List<ClassMemberAstNode>();

    public static ClassBodyAstNode Parse(Parser parser)
    {
        ClassBodyAstNode node = new ClassBodyAstNode();

        parser.Take(TokenType.OPEN_CURLY);

        while(true) {
            ClassMemberAstNode? memberNode = ClassMemberAstNode.TryParse(parser);
            if (memberNode == null) break;
            node.memberList.Add(memberNode);
        }

        return node;
    }
}

public class ClassMemberAstNode
{
    public static ClassMemberAstNode? TryParse(Parser parser)
    {
        ClassMemberAstNode node = new ClassMemberAstNode();

        return node;
    }
}